python/ctsm/base_case.py

```diff
@@ -0,0 +1,127 @@
+import os
+import numpy as np
+import xarray as xr
+import subprocess 
+
+from datetime import date
+from getpass import getuser
+
+myname = getuser() 
```

Comment:
Update: here's a reference: https://google.github.io/styleguide/pyguide.html#25-global-variables

Reply:
Thanks for this comment. I myself am not a big fan of global variables and try to avoid them as much as possible... But as you said, it is sometimes necessary to have a few around....

In this special case, `myname` is a global variable because the argument parser needs it immediately and it also needed in the main... The argument parser needs it because it creates the default paths for different users based on the usernames... For example, if you run this code, the default path that it shows for you when you run `./subset_data point --help` points to your scratch while for me it is pointing to mine...

What should one do in these situations? I am using this implementation for `myname` and similar implementation for `my_machine`  in other codes too. So any advice on these will be appreciated...